PHP: add use_tls and use_test_ca param to interop tests script

stanley-cheung opened the pull request:
* Fixes #3732 
 * And some general clean up

stanley-cheung commented:
Main jenkins build is green. Interop build has the known errors.

jtattermusch commented:
LGTM 
Can you also update PHPLanguage in run_interop_tests.py to use the --use_tls=true and --use_test_ca=true. It's fine to do that in a followup PR if you prefer to.

stanley-cheung commented:
I added the `--use_tls=true` flag to `run_interop_tests.py`. And I also added some code to `run_interop_tests.py` to bring the `language.global_env()` environment variables to the `build_interop_image.sh` job. This is to further automate #3818 a bit instead of relying on user to pass that variable from the command line. Jan, can you please take a look at the last 3 commits? Thanks.

stanley-cheung commented:
The interop tests Jenkins build is looking good for PHP. The 2 auth interop tests failure are addressed in the next PR #3854.

jtattermusch commented on tools/run_tests/run_interop_tests.py:
```diff
@@ -220,17 +220,20 @@ def __init__(self):
 
   def global_env(self):
-    return {}
+    # need to manually copy to each jenkins machine if we run into github
```
Currently, global_env (as well as all other *_env  functions) are used to set the environment variable setting to actually run the interop client and interop server (see python), not for running the build_interop_image.sh script. 

So what you are doing here make usage of global_env pretty inconsistent  and I would be against it.

stanley-cheung replied:
Ah, ic. I will address this in a new PR. Thanks for the background.

jtattermusch commented:
Alright, by objection came after it was merged. Can we remove the changes from last two commits?

stanley-cheung commented:
Sure. I will do it in a new PR.
